### ISSUE TITLE
grayed area in column mode

### ISSUE THREAD

**Issue by dracodoc** [koreader/kindlepdfviewer#168]
I'm reading a 2 column pdf (which is a Acrobat OCRed cleanscan file). In column mode, when I press down with the five way rocker, the lower part of column have the upper area(6 lines) grayed. If I press up, the higher part of column have the lower area grayed too, even at first the higher part was displayed correctly. Press F again will back to left corner and reset the page, and the gray area will gone.

The current version don't have the screen shot function yet, right? If you need, I can take the photo of the screen, but I think the description should be enough to reproduce the problem. Or it could be related to the pdf, I uploaded a part of the pdf here
https://www.box.com/s/3334a97936f4fa39f923

I noticed the memory usage is larger than the limit, document memory was 10312 over the 8192 limit, so I increased it to 16M, but the problem was still there.

If press menu, then back, the screen will refresh, and the gray area will gone.

**Comment by dpavlin** [comment]
You can use ssh and dd to take screenshots. Example of such script is https://github.com/dpavlin/k3libre/blob/master/ssh-screenshot.sh

I did tweak margin code quite a few times, and I only graying I can reproduce with current master is in shift+F mode, on 600*800 screen when I press down fiveway twice on first page. But, I'm not sure if that is effect you are describing.

**Comment by dracodoc** [comment]
I took some pics and uploaded here, start from this one
https://picasaweb.google.com/lh/photo/YvRKG2nNIGf2KLVXSbIYyNMTjNZETYmyPJy0liipFm0?feat=directlink

https://picasaweb.google.com/lh/photo/zBv56FYoGOkKhzS-wvLBytMTjNZETYmyPJy0liipFm0?feat=directlink

https://picasaweb.google.com/lh/photo/KsP0eYlIkVcQIxM6BCmpWtMTjNZETYmyPJy0liipFm0?feat=directlink

https://picasaweb.google.com/lh/photo/uR1WwlEKoxwoAYygBPq7wdMTjNZETYmyPJy0liipFm0?feat=directlink

It doesn't look like margin. The gray area could disappear when press menu then back, this could be a hint, why will press menu and back will make gray area disappear but Shift R will not?

**Comment by dpavlin** [comment]
Oh, now I understand. It's behaving as it should :-) Idea behind gray-out part of screen to to mark part of page which you already seen so that's why this part is always grayed out.

Shift+R will just refresh e-ink display while entering and exiting menu will redraw whole page and reset grayed out margin. I find difference useful if I want to remove grayed out part.

**Comment by dracodoc** [comment]
OK, got it. Actually I thought about this when I first saw it. However when I move back to previous view by press up, the lower part of upper column was gray-out too. I think here it should be displayed as fully normal, because this gray part should at least displayed as normal in one of two views.
https://picasaweb.google.com/lh/photo/uR1WwlEKoxwoAYygBPq7wdMTjNZETYmyPJy0liipFm0?feat=directlink

So all the other steps are normal, just the pic above is not quite right.
In other words:
press down move to lower part
press up to move back to upper part, here the full screen should be normal.

**Comment by dpavlin** [comment]
Current code simply displays already displayed part as grayed out. Implementing graying that you suggest would involve some kind of additional storage which would record last move so we can detect this special case (up/down) and would introduce additional code complexity while violating original design of "always show already visible part as grayed out".

Having said that, I'm of course biased toward current implementation since I wrote it :-) I tend to read full page including grayed out part, so graying it out again make sense to me since it indicates point in which I should continue reading.

**Comment by dracodoc** [comment]
OK, I can accept current design. At first I was confused about the gray part because it appeared in previous view too. Knowing the meaning of it I can understand and accept it.
